nym-data-observatory/build.rs

```diff
@@ -7,41 +7,44 @@ const POSTGRES_USER: &str = "nym";
 const POSTGRES_PASSWORD: &str = "password123";
 const POSTGRES_DB: &str = "data_obs_db";
 
-/// If you need to re-run migrations or reset the db, just run
-/// cargo clean -p nym-node-status-api
+/// if schema changes, rerun `cargo sqlx prepare` with a running DB
+/// https://github.com/launchbadge/sqlx/blob/main/sqlx-cli/README.md#enable-building-in-offline-mode-with-query
 #[tokio::main]
 async fn main() -> Result<()> {
-    if let Ok(value) = std::env::var("CI") {
-        if value == "true" {
-            println!("cargo::rustc-env=SQLX_OFFLINE=true");
-        }
-    } else {
-        let db_url = export_db_variables()?;
+    let db_url = format!(
+        "postgresql://{}:{}@localhost:5432/{}",
+        POSTGRES_USER, POSTGRES_PASSWORD, POSTGRES_DB
+    );
+
+    // if a live DB is reachable, use that
+    if PgConnection::connect(&db_url).await.is_ok() {
+        export_db_variables(&db_url)?;
```

Comment:
can we actually connect to the db BEFORE exporting those variables?

Reply:
here, yes, because their values are embedded in a db_url string

env vars exported so rust-analyzer doesn't complain